WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt

```diff
@@ -0,0 +1,145 @@
+@file:JvmName("SupportHelper")
+
+package org.wordpress.android.support
+
+import android.content.Context
+import android.support.v7.app.AlertDialog
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.widget.EditText
+import android.widget.TextView
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.AccountModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.prefs.AppPrefs
+import org.wordpress.android.util.validateEmail
+
+/**
+ * This function will check whether there is a support email saved in the AppPrefs and use the saved email and name
+ * to run the provided function, most likely a support request. If there is no saved support email, it'll trigger a
+ * function to show a dialog with email and name input fields which then it'll save to AppPrefs and run the provided
+ * function.
+ *
+ * @param context Context the dialog will be showed from
+ * @param account WordPress.com account to be used for email and name suggestion in the input dialog
+ * @param selectedSite Selected site to be used for email and name suggestion in case the user is not logged in
+ * @param emailAndNameSelected Function to run with the email and name from AppPrefs or the input dialog
+ */
+fun showSupportIdentityInputDialogAndRunWithEmailAndName(
+    context: Context,
+    account: AccountModel?,
+    selectedSite: SiteModel?,
+    emailAndNameSelected: (String, String) -> Unit
+) {
+    val currentEmail = AppPrefs.getSupportEmail()
+    if (!currentEmail.isNullOrEmpty()) {
+        emailAndNameSelected(currentEmail, AppPrefs.getSupportName())
+    } else {
+        val (emailSuggestion, nameSuggestion) = supportEmailAndNameSuggestion(account, selectedSite)
+        showSupportIdentityInputDialog(context, emailSuggestion, nameSuggestion, false) { email, name ->
+            AppPrefs.setSupportEmail(email)
+            AppPrefs.setSupportName(name)
+            emailAndNameSelected(email, name)
+        }
+    }
+}
+
+/**
+ * This is a helper function that shows the support identity input dialog and runs the provided function with the input
+ * from it.
+ *
+ * @param context Context the dialog will be showed from
+ * @param email Initial value for the email field
+ * @param name Initial value for the name field
+ * @param isNameInputHidden Whether the name input field should be shown or not
+ * @param emailAndNameSelected Function to run with the email and name inputs from the dialog. Even if the
+ * [isNameInputHidden] parameter is true, the input in the name field will be provided and it's up to the caller to
+ * ignore the name parameter.
+ */
+fun showSupportIdentityInputDialog(
```

Comment:
I thought I'd use this function directly in the upcoming PR, but I since decided to keep it all in the `SupportHelper`. Long story short, it'll be `private`.